### ISSUE TITLE
upgrade request 1.18.2

### ISSUE THREAD

**Issue by ptk2003** [rikka0w0/SimElectricity#18]
I can build jar from source 1.18.1 but it not loading in 1.18.2 mc

**Comment by AuroraRAS** [comment]
Can you provide more details? like some errors line in your game launcher.

**Comment by ptk2003** [comment]
Mod File: SimElectricity-1.0.0.jar
	Failure message: SimElectricity Essential (sime_essential) has failed to load correctly
		java.lang.NoClassDefFoundError: net/minecraftforge/event/RegistryEvent$NewRegistry
	Mod Version: 1.0.0
	Mod Issue URL: https://github.com/RoyalAliceAcademyOfSciences/SimElectricity/issues
	Exception message: java.lang.ClassNotFoundException: net.minecraftforge.event.RegistryEvent$NewRegistry

[latest.log](https://github.com/RoyalAliceAcademyOfSciences/SimElectricity/files/8645368/latest.log)
[crash-2022-05-07_17.28.06-fml.txt](https://github.com/RoyalAliceAcademyOfSciences/SimElectricity/files/8645369/crash-2022-05-07_17.28.06-fml.txt)
![2022-05-07 17_28_19-Minecraft_ 1 18 2](https://user-images.githubusercontent.com/28514017/167258738-9bc907d3-54f7-4828-85c7-09b73b9d2232.png)

**Comment by AuroraRAS** [comment]
Maybe you need redownload forge and its library? I guess.
enable your VPN if you are living in China.

**Comment by ptk2003** [comment]
I download and install latest 1.18.2-forge-40.1.16 it not solve issue (same error), all other mods build for 1.18.2 loading properly (they are removed for test to prevent mod incompatible).

Wich version of forge you can load mod (tested)?

**Comment by AuroraRAS** [comment]
Hey, @rikka0w0 ! We need some help.

**Comment by ptk2003** [comment]
Hi. Any news?

**Comment by rikka0w0** [comment]
That's a known problem, 1.18.1->1.18.2 isn't a minor update, despite the gameplay does not change at all, things have changed quite a lot internally. I'm working on a total re-write of the registry stuff and it will fix this issue. Be patient. If you are also a developer, any help would be appreciated.

**Comment by rikka0w0** [comment]
Basically, the problem originates from the instantiation of registrable objects. Before 1.18.2, you can instantiate those things at any time before the actual registration happens, so we chose to initialize them statically. But since 1.18.2, you have to instantiate them at the correct instant, just before the registration event is fired.

**Comment by AuroraRAS** [comment]
We have upgraded the current version to `NeoForge 1.21.1`.

**Comment by ptk2003** [comment]
The PowerGrid mod was recently released for 1.20.1 and 1.21.1. 
It is fully compatible with the Rotational Energy mod from Create and the Forge energy converter, so my issue is no longer relevant.

And I couldn't even test your new version!
The instructions on GitHub don't work!

##############
 1   Ensure Java (found here) and Git (found here) are properly installed on your system.
 2   Create a base directory for the repo (anywhere you like)
 3   On Windows, open either 'CMD' or Windows PowerShell, on Linux and MacOS, launch a terminal, then navigate to the directory just created, and type the following commands:
 4   git clone https://github.com/RoyalAliceAcademyOfSciences/SimElectricity
 5   git submodule init and git submodule update to get LibRikka

 6   On Windows: use gradlew.bat instead of gradlew
##############

1. Your link suggests downloading version jre-8u491-windows-x64.exe. Ok, we downloaded it (even though it's very outdated and doesn't work anywhere).
2, 3, 4, - okay

_5. Error:
fatal: not a git repository (or any of the parent directories): .git
because you missed the command to go to the SimElectricity folder.

OK, fixed.
6. gradlew.bat opens and immediately closes because someone  include "exit /b 1" in the error section.
Okay, I fixed it, but when I run it, I get the error:

FAILURE: Build failed with an exception.

* What went wrong:
Gradle requires JVM 17 or later to run. Your build is currently configured to use JVM 8.

--
Basically, the instructions for programmers are something I didn't understand... why can't you just post the jar file on CourseForge like other mods ???

**Comment by AuroraRAS** [comment]
I apologize, but as you can see, this project has just revived after several years of dormancy and is currently in an adaptation phase before resuming its official mod release process.

I've just built these [three files](https://github.com/rikka0w0/SimElectricity/releases/tag/1.21.1), which you can use for testing. Due to a lack of testing environment, I don't know if they will load correctly. However, I assure you that it will indeed run correctly in the development environment.

<img width="954" height="654" alt="Image" src="https://github.com/user-attachments/assets/2a4e5b8e-7340-4406-b53a-2695618b4c67" />

The old version of `gradlew.bat` has been updated; please try pulling it.

**Comment by ptk2003** [comment]
Download files, thank you
